src/redux/propertytree/propertyTreeSlice.ts

```diff
@@ -0,0 +1,194 @@
+// removeProperties
+// updatePropertyValue
+// setPropertyValue
+// clearPropertyTree
+
+export interface PropertyOwnersState {
+  isInitialized: boolean;
+  propertyOwners: PropertyOwners;
+}
+
+const initialStatePropertyOwners: PropertyOwnersState = {
+  isInitialized: false,
+  propertyOwners: {}
+};
+
+export const propertyOwners = createSlice({
+  name: 'propertyOwners',
+  initialState: initialStatePropertyOwners,
+  reducers: {
+    addPropertyOwners: (
+      state,
+      action: PayloadAction<{ propertyOwners: PropertyOwner[] }>
+    ) => {
+      const inputOwners = action.payload.propertyOwners;
+      // const newState = { ...state };
+      inputOwners.forEach((owner) => {
+        state.propertyOwners[owner.uri] = {
+          identifier: owner.identifier,
+          name: owner.name,
+          properties: owner.properties,
+          subowners: owner.subowners,
+          tags: owner.tags || [],
+          uri: owner.uri, // TODO: Remove this
+          description: owner.description
+        };
+
+        // Ensure the parents of the uri have the links to the new entry
+        // Get parent uri
+        const periodPos = owner.uri.lastIndexOf('.');
+        const parentUri = owner.uri.substring(0, periodPos);
+
+        // If that parent exists and the link doesn't exist, add the link to the parent
+        const parentExists = parentUri && state.propertyOwners[parentUri];
+        if (
+          parentExists &&
+          !state.propertyOwners[parentUri]!.subowners.includes(owner.uri)
+        ) {
+          state.propertyOwners[parentUri]!.subowners.push(owner.uri);
+        }
+      });
+      return state;
+    },
+    clearPropertyTree: () => {
+      return {
+        isInitialized: false,
+        propertyOwners: {}
+      };
+    },
+    removePropertyOwners: (state, action: PayloadAction<{ uris: string[] }>) => {
+      // const newState = { ...state };
+
+      action.payload.uris.forEach((uri) => {
+        // Delete this particular property owner
+        delete state.propertyOwners[uri];
+
+        // Delete the parent's link to the property owner
+        const periodPos = uri.lastIndexOf('.');
+        const parentUri = uri.substring(0, periodPos);
+        const index = state.propertyOwners[parentUri]?.subowners.indexOf(uri) ?? -1;
+        // If found, remove
+        if (index > -1) {
+          // 2nd parameter means remove one item only
+          state.propertyOwners[parentUri]!.subowners.splice(index, 1);
+        }
+
+        // Delete subowners that have been flattened
+        const related = Object.keys(state).filter((value) => value.includes(`${uri}.`));
+        related.forEach((subOwnerUri) => delete state.propertyOwners[subOwnerUri]);
+      });
+      return state;
+    }
+  }
+});
+
+export const { addPropertyOwners, removePropertyOwners } = propertyOwners.actions;
+export const propertyTreeReducer = combineSlices({
+  props: properties.reducer,
+  owners: propertyOwners.reducer
```

Comment:
In particular, I do not think we should use the `props` shortening here (since this can be misinterpreted with the redux props). 

But I also think we should avoid renaming these states, in general. It's better to be too specific rather than vague in the naming, and not renaming them will simplify for us when implementing the existing components



```suggestion
  properties: properties.reducer,
  propertyOwners: propertyOwners.reducer
```

Reply:
Yeah I agree, what I did not like about having their original names was that when we access the state `properties` or `propertyOwners` it will be written twice for example here: `(state) => state.propertyTree.props.properties[NavigationAimKey]` vs ` (state) => state.propertyTree.properties.properties[NavigationAimKey]`

The combineSlices or combineReducers does not seem to work as they did before where it was all smudged into one reducer. So one alternative could be that we write them as completely separate reducers, then you would access them as `state.properties[key]` and `state.propertyOwner[key]`, though it is nice that they are in the `propertyTree` subobject so I'm not too sure about this either..